### PR TITLE
reuse cdn sample code and version

### PR DESCRIPTION
**Pull request by mifi**
see #15

**don't merge before #27**

### CODE REVIEW

File: src/components/UppyCdnExample.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+<link href="https://releases.transloadit.com/uppy/v${uppyVersion}/${uppyCssName}" rel="stylesheet">
+
+<!-- 2. Add JS before the closing \`</body>\` -->
+<script src="https://releases.transloadit.com/uppy/v${uppyVersion}/${uppyJsName}"></script>
+
+<!-- 3. Initialize -->
+<div id="uppy"></div>
+
+<script>
+  var uppy = new Uppy.Core()${linesProcessed.length > 0 ? `\n\n${linesProcessed}` : ''}
```
Comment (Murderlon) [inline src/components/UppyCdnExample.tsx:32]: We shouldn't put uppy core here by default, I think it's nicer to see the complete example while writing docs and we may want to change the options for some examples.

File: src/components/UppyCdnExample.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export default function UppyCdnExample ({
+  children,
+  uppyCssName = 'uppy.min.css',
+  uppyJsName = 'uppy.min.js',
+}) {
+  let lines = [];
+
+  React.Children.toArray(children).forEach((child) => {
+    lines = [...lines, ...String(child).trim().split('\n').map((line) => line.trim())];
+  })
```
Comment (Murderlon) [inline src/components/UppyCdnExample.tsx:17]: I'm surprised this is necessary, if we pass a correct string with newlines and everything, won't it look good?

**Review by Murderlon — COMMENTED** [review]
Nice!

File: src/components/UppyCdnExample.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import CodeBlock from '@theme/CodeBlock';
+import Admonition from '@theme/Admonition';
+
+
+const uppyVersion = '2.9.0' // todo
```
[the remark this answers is not in the document]
Comment (Murderlon) [reply src/components/UppyCdnExample.tsx:6]: Note that we may already have these hooks in our deps

File: src/components/UppyCdnExample.tsx (hunk 1 above)
Comment (mifi) [reply src/components/UppyCdnExample.tsx:32]: What should we use instead? I see all the existing examples use Uppy.Core.

Comment (mifi) [reply src/components/UppyCdnExample.tsx:32]: Oh you mean we should leave the script tag empty

File: src/components/UppyCdnExample.tsx (hunk 2 above)
Comment (mifi) [reply src/components/UppyCdnExample.tsx:17]: Yes, but I think it's nice if the developer doesn't have to worry about providing the correct indentation everytime we want to use the component.

File: src/components/UppyCdnExample.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mifi) [reply src/components/UppyCdnExample.tsx]: Couldn't get that hook to work but I used `import packageJson from '../../package.json'` and it seems to work

File: src/components/UppyCdnExample.tsx (hunk 1 above)
Comment (Murderlon) [reply src/components/UppyCdnExample.tsx]: No I meant it feels less magical to explicitly write an example every time, so I personally wouldn't put the default uppy core in here already

File: docs/sources/companion-plugins/url.mdx [hunk 4]
```diff
@@ -42,27 +43,14 @@ so you don’t have to worry about the security implications of arbitrary URLs.
...
-      const Url = Uppy.Url
-    </script>
-  ```
-
+    <UppyCdnExample>
+      {`
+        var uppy = new Uppy.Core()
+        uppy.use(Uppy.Url, {
+          // Options
+        })
```
Comment (Murderlon) [inline docs/sources/companion-plugins/url.mdx:51]: Everything seems to have an extra newline in between. Is the output different?

**Review by Murderlon — APPROVED** [review]

File: docs/sources/companion-plugins/url.mdx (hunk 4 above)
Comment (Murderlon) [reply docs/sources/companion-plugins/url.mdx:51]: Never mind it was GitHub's weird diff

**Comment by aduh95** [comment]
`Uppy.Core` no longer exists in Uppy 3.x, now it's `Uppy.Uppy`.

**Comment by Murderlon** [comment]
I forgot indeed that we now write out CDN example like so (`type="module"` and import from URL):

```js
<link href="https://releases.transloadit.com/uppy/v3.0.1/uppy.min.css" rel="stylesheet">

<div id="uppy"></div>

<script type="module">
  import { Uppy, Dashboard, Tus } from "https://releases.transloadit.com/uppy/v3.0.1/uppy.min.mjs"

  const uppy = new Uppy()
  uppy.use(Dashboard, { target: '#files-drag-drop' })
  uppy.use(Tus, { endpoint: 'https://tusd.tusdemo.net/files/' })
</script>
```

@mifi not sure how we can easily reference the correct version when we still write it everywhere instead of in the component?

**Comment by mifi** [comment]
Not sure, but maybe we need to use some placeholder for interpolation? something like `v{{UPPY_VERSION}}` and do a `String(children).replace(/{{UPPY_VERSION}}/g, uppyVersion)`

**Comment by Murderlon** [comment]
Feels a bit dirty but that might indeed be the only alternative to running a script which does the same.

**Comment by Murderlon** [comment]
Would you be willing to give that a try? :)

**Pull request merged**